### PR TITLE
[TC-807] Scheduler Changes

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0807](https://bcdevex.atlassian.net/browse/TC-0807)

Objective:
- remove 'AVAILABLE' and 'NOT_INDICATED' from availability type enum
- instead use null and delete if updating from deployed/unavailable to available
- show white on member profile to indicate available and green otherwise

*note:* currently there is a bug when filtering the dash by availability. We are only returning full, not partial matched. Have left a TODO. 

[Ticket to fix filter](https://bcdevex.atlassian.net/browse/TC-821)

### CODE REVIEW

File: backend/src/bcws/bcws.service.ts [hunk 1]
```diff
@@ -161,14 +162,15 @@ export class BcwsService {
    * @returns {number} Count of total personnel search applies to
    */
   async getBcwsPersonnel(query: GetBcwsPersonnelDTO): Promise<{
-    personnel: BcwsPersonnelEntity[];
+    personnel: BcwsPersonnelEntity[]|PersonnelEntity[];
```
Comment (fw-noel) [inline backend/src/bcws/bcws.service.ts:165]: Mind explaining this one?

File: backend/src/database/entities/availability.entity.ts [hunk 2]
```diff
@@ -25,9 +25,10 @@ export class AvailabilityEntity {
     type: 'enum',
     enum: AvailabilityType,
     enumName: 'availability-type',
-    default: AvailabilityType.NOT_INDICATED,
+    nullable: true,
+    default: null
```
Comment (fw-noel) [inline backend/src/database/entities/availability.entity.ts:29]: Missing trailing comma, check a prettier

File: backend/src/database/entities/personnel.entity.ts [hunk 3]
```diff
@@ -166,17 +167,20 @@ export class PersonnelEntity extends BaseEntity {
       availability:
         this.availability?.map((avail) => avail.toResponseObject()) || [],
     };
-
     // If availability is empty (hence we requested it and nothing came back), we fill in a "today"
-    if (this.availability && data.availability.length === 0) {
-      data.availability = [
-        new AvailabilityEntity({
+    // console.log(data.availability)  
+    if (data.availability.length === 0 || data.availability === undefined
```
Comment (fw-noel) [inline backend/src/database/entities/personnel.entity.ts:172]: Weird spacing

Comment (fw-noel) [reply backend/src/database/entities/personnel.entity.ts:172]: Also a log above

File: backend/src/personnel/personnel.service.ts [hunk 4]
```diff
@@ -165,14 +176,74 @@ export class PersonnelService {
...
+      const partialAvailableQuery = this.availabilityRepository.createQueryBuilder('availability');
+      partialAvailableQuery.select('personnel')
+      partialAvailableQuery.where('availability.date >= :from AND availability.date <= :to', {
+        from: new Date(availabilityFromDate),
+        to: new Date(availabilityToDate),
+      })
+      partialAvailableQuery.groupBy('personnel')
+      partialAvailableQuery.having(`COUNT(*) < :count`, { count: numberOfDays });
+      
+      const partAvailable = this.personnelRepository.createQueryBuilder('personnel');
```
Comment (fw-noel) [inline backend/src/personnel/personnel.service.ts:199]: Is this actually being used? Mind cleaning up with some comments?

File: backend/src/emcr/emcr.service.ts [hunk 5]
```diff
@@ -126,7 +127,7 @@ export class EmcrService {
    * @returns {number} Count of total personnel search applies to
    */
   async getEmcrPersonnel(query: GetEmcrPersonnelDTO): Promise<{
-    personnel: EmcrPersonnelEntity[];
+    personnel: EmcrPersonnelEntity[] | PersonnelEntity[];
```
Comment (fw-noel) [inline backend/src/emcr/emcr.service.ts:130]: You removed the equivalent from bcws service, correct?

**Review by fw-noel — APPROVED** [review]

File: backend/src/emcr/emcr.service.ts (hunk 5 above)
Comment (chelsea-EYDS) [reply backend/src/emcr/emcr.service.ts:130]: oh shoot one second I'll remove from here

**Review by fw-noel — APPROVED** [review]

**Pull request merged**